packages/plugin/src/schema.ts

```diff
@@ -32,25 +33,23 @@ export function getSchema(options: ParserOptions, gqlConfig: GraphQLConfig): Gra
 
   // We first try to use graphql-config for loading the schema, based on the type of the file,
   // We are using the directory of the file as the key for the schema caching, to avoid reloading of the schema.
-  if (gqlConfig && options?.filePath) {
-    const fileDir = dirname(options.filePath);
+  if (gqlConfig && options.filePath) {
+    const realFilepath = getOnDiskFilepath(options.filePath);
+    const fileDir = dirname(realFilepath);
 
     if (schemaCache.has(fileDir)) {
       schema = schemaCache.get(fileDir);
     } else {
-      const projectForFile = gqlConfig.getProjectForFile(options.filePath);
-
-      if (projectForFile) {
```

Comment:
`projectForFile` is always truthy based on type definition. And I think this will be great if we add` "strict": true` to our `tsconfig.json`.